<a href="https://colab.research.google.com/github/JoeBwonKenobi/Car-Insurance-Project/blob/main/Project2_Part_1ipynb.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Car Insurance Prediction Project**

Joe Lardie

Jan 2023

##**Overview Of Project**

- For this Task:
Create a Colab notebook where you have uploaded and shown the .head() of each of your data sets. For each of the proposed datasets, answer the following questions:

First choice: dataset 1

1. Source of data

2. Brief description of data

3. What is the target?

4. What does one row represent? (A person?  A business?  An event? A product?)

5. Is this a classification or regression problem?

6. How many features does the data have?

7. How many rows are in the dataset?

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?

Second choice: dataset 2

1. Source of data

2. Brief description of data

3. What is the target?

4. What does one row represent? (A person?  A business?  An event? A product?)

5. Is this a classification or regression problem?

6. How many features does the data have?

7. How many rows are in the dataset?

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?

- Link to the original data sources on Kaggle:

https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset


https://www.kaggle.com/datasets/sagnik1511/car-insurance-data

##**Imports**

In [1]:
#Numpy
import numpy as np

#Pandas
import pandas as pd

#Seaborn
import seaborn as sns

#matplotlib
import matplotlib.pyplot as plt

##**Loading Data**

In [2]:
from google.colab import drive
drive.mount('/content/drive/')

Drive already mounted at /content/drive/; to attempt to forcibly remount, call drive.mount("/content/drive/", force_remount=True).


In [3]:
# Load Data for Car Insurance Data
cid = '/content/drive/MyDrive/CodingDojo/Colab Notebooks/Stack 2/Data/Car Insurance Data.csv'
df = pd.read_csv(cid)
df.head()

ID    AGE  GENDER      RACE DRIVING_EXPERIENCE    EDUCATION  \
0  569520    65+  female  majority               0-9y  high school   
1  750365  16-25    male  majority               0-9y         none   
2  199901  16-25  female  majority               0-9y  high school   
3  478866  16-25    male  majority               0-9y   university   
4  731664  26-39    male  majority             10-19y         none   

          INCOME  CREDIT_SCORE  VEHICLE_OWNERSHIP VEHICLE_YEAR  MARRIED  \
0    upper class      0.629027                  1   after 2015        0   
1        poverty      0.357757                  0  before 2015        0   
2  working class      0.493146                  1  before 2015        0   
3  working class      0.206013                  1  before 2015        0   
4  working class      0.388366                  1  before 2015        0   

   CHILDREN  POSTAL_CODE  ANNUAL_MILEAGE VEHICLE_TYPE  SPEEDING_VIOLATIONS  \
0         1        10238         12000.0        sedan                    0   
1         0        10238         16000.0        sedan                    0   
2         0        10238         11000.0        sedan                    0   
3         1        32765         11000.0        sedan                    0   
4         0        32765         12000.0        sedan                    2   

   DUIS  PAST_ACCIDENTS  OUTCOME  
0     0               0        0  
1     0               0        1  
2     0               0        0  
3     0               0        0  
4     0               1        1

In [4]:
cid = df.copy()

In [5]:
cid.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 19 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID                   10000 non-null  int64  
 1   AGE                  10000 non-null  object 
 2   GENDER               10000 non-null  object 
 3   RACE                 10000 non-null  object 
 4   DRIVING_EXPERIENCE   10000 non-null  object 
 5   EDUCATION            10000 non-null  object 
 6   INCOME               10000 non-null  object 
 7   CREDIT_SCORE         9018 non-null   float64
 8   VEHICLE_OWNERSHIP    10000 non-null  int64  
 9   VEHICLE_YEAR         10000 non-null  object 
 10  MARRIED              10000 non-null  int64  
 11  CHILDREN             10000 non-null  int64  
 12  POSTAL_CODE          10000 non-null  int64  
 13  ANNUAL_MILEAGE       9043 non-null   float64
 14  VEHICLE_TYPE         10000 non-null  object 
 15  SPEEDING_VIOLATIONS  10000 non-null  

In [6]:
cid.isna().sum()

ID                       0
AGE                      0
GENDER                   0
RACE                     0
DRIVING_EXPERIENCE       0
EDUCATION                0
INCOME                   0
CREDIT_SCORE           982
VEHICLE_OWNERSHIP        0
VEHICLE_YEAR             0
MARRIED                  0
CHILDREN                 0
POSTAL_CODE              0
ANNUAL_MILEAGE         957
VEHICLE_TYPE             0
SPEEDING_VIOLATIONS      0
DUIS                     0
PAST_ACCIDENTS           0
OUTCOME                  0
dtype: int64

In [7]:
# Load Stroke Dahealthcare-dataset-stroke-data.csvta
stk = '/content/healthcare-dataset-stroke-data.csv'
df_s = pd.read_csv(stk)
df_s.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [8]:
df_s.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


In [9]:
df_s.isna().sum()

id                     0
gender                 0
age                    0
hypertension           0
heart_disease          0
ever_married           0
work_type              0
Residence_type         0
avg_glucose_level      0
bmi                  201
smoking_status         0
stroke                 0
dtype: int64

##**Description of Data**

###**Car Insurance Data Set**

- The Car Insurance Data Describes information about different car insurance customer's behavior. The different behavior statistics will help us classify which behaviors can help predict if they make a claim or not.

 - The different rows in the data are different insurance customers, and the columns are things that may effect if a customer makes a claim or not.

  - For this data set we will use classification metrics to take a look at the different things that my attribute to someone having to file a claim for their insurance.

  - There are 19 total features in this data set, our target is the Outcome column.

  - In this data set there are 10,000 rows (customers).

  - There are a few inconsistancies within this data set, that we will have to take care of before it is ready for modeling.There is missing data in the 'Credit score' column and the 'Annual mileage' column. We can impute data here before our data is readay to do modeling.
  - The 'ID' column could potentially be dropped as it serves no purpose in aiding us to predict the cost of someone's insurance.
  - To properly prepare this data set for machine learning, there are eight columns that need to be converted from an object data type into a numeric data type. These are as follows: Age, Gender, Race, Driving Expierence, Education, Income, Vehicle Year, and Vehicle Type.

###**Stroke Data Set**

- The Stroke data set provides information relating to things that may increase the chance of someone having a stroke.

 - The different rows in the data are different patients, with the columns being different categories of things that could influence a person's chances of having a stroke.

  - With this data set we'll use regression metrics to take a look at the things that can help predict if someone may be at risk for having a stroke.

  - There are 12 total features in this data set, our target is the stroke column.

  - In this data set there are 5,110 rows (patients).

  - There are a few inconsistancies with the stroke data set. The BMI column has missing values. We can impute data here before our data is readay to do modeling. 
  
  -This data set also has an 'ID' column that serves us no purpose in finding the probability of someone having a stroke.

  - To properly prepare this data set for machine learning, there are five columns that need to be converted from an object data typr into a numeric data type. These are as follows: Gender, Ever Married, Work type, Residence Type, and smoking statis.